### ISSUE TITLE
Why Leiden algorithm in FindCluster only identify one final cluster after trying many times??

### ISSUE THREAD

**Issue by Sophia409** [satijalab/seurat#2817]
Hello,

I can use default Louvain algorithm to get right numbers of clusters, but it failed when I tried Leiden algorithm. Though I adjusted the resolution to a larger value, it still only produced one final cluster.

I'm really confused. And the Seurat version I used is 3.1.2. 

Hope for your help.
```
Group <- FindClusters(object = Group,resolution=1,algorithm = 4)
7227 singletons identified. 1 final clusters.
Warning: Adding a command log without an assay associated with it

Group
An object of class Seurat 
30879 features across 7230 samples within 2 assays 
Active assay: integrated (2000 features)
 1 other assay present: RNA
 3 dimensional reductions calculated: pca, tsne, umap

sessionInfo()
R version 3.6.3 (2020-02-29)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18362)

Matrix products: default

locale:
[1] LC_COLLATE=Chinese (Simplified)_China.936 
[2] LC_CTYPE=Chinese (Simplified)_China.936   
[3] LC_MONETARY=Chinese (Simplified)_China.936
[4] LC_NUMERIC=C                              
[5] LC_TIME=Chinese (Simplified)_China.936    

attached base packages:
 [1] splines   stats4    parallel  stats     graphics  grDevices utils     datasets 
 [9] methods   base     

other attached packages:
 [1] loomR_0.2.1.9000            hdf5r_1.3.1                
 [3] R6_2.4.1                    doRNG_1.8.2                
 [5] rngtools_1.5                doParallel_1.0.15          
 [7] iterators_1.0.12            foreach_1.4.8              
 [9] SCopeLoomR_0.5.0            RcisTarget_1.6.0           
[11] AUCell_1.8.0                SCENIC_1.1.2-2             
[13] openxlsx_4.1.4              data.table_1.12.8          
[15] enrichplot_1.6.1            gridExtra_2.3              
[17] gtools_3.8.1                DoubletFinder_2.0.2        
[19] org.Mm.eg.db_3.10.0         AnnotationDbi_1.48.0       
[21] clusterProfiler_3.14.3      scales_1.1.0               
[23] ggThemeAssist_0.1.5         scater_1.14.6              
[25] ggsci_2.9                   ape_5.3                    
[27] pheatmap_1.0.12             plyr_1.8.5                 
[29] cowplot_1.0.0               slingshot_1.4.0            
[31] princurve_2.1.4             monocle3_0.2.1             
[33] SingleCellExperiment_1.8.0  SummarizedExperiment_1.16.1
[35] DelayedArray_0.12.2         BiocParallel_1.20.1        
[37] matrixStats_0.55.0          GenomicRanges_1.38.0       
[39] GenomeInfoDb_1.22.0         IRanges_2.20.2             
[41] S4Vectors_0.24.3            monocle_2.14.0             
[43] DDRTree_0.1.5               irlba_2.3.3                
[45] VGAM_1.1-2                  Biobase_2.46.0             
[47] BiocGenerics_0.32.0         Matrix_1.2-18              
[49] RColorBrewer_1.1-2          corrplot_0.84              
[51] set_1.1                     ggcorrplot_0.1.3           
[53] forcats_0.5.0               stringr_1.4.0              
[55] dplyr_0.8.4                 purrr_0.3.3                
[57] readr_1.3.1                 tidyr_1.0.2                
[59] tibble_2.1.3                ggplot2_3.2.1              
[61] tidyverse_1.3.0             leiden_0.3.3               
[63] Seurat_3.1.2                reticulate_1.14            

loaded via a namespace (and not attached):
  [1] rsvd_1.0.3               ica_1.0-2                ps_1.3.2                
  [4] lmtest_0.9-37            rprojroot_1.3-2          crayon_1.3.4            
  [7] MASS_7.3-51.5            nlme_3.1-144             backports_1.1.5         
 [10] qlcMatrix_0.9.7          reprex_0.3.0             GOSemSim_2.12.0         
 [13] rlang_0.4.4              XVector_0.26.0           ROCR_1.0-7              
 [16] readxl_1.3.1             callr_3.4.2              limma_3.42.2            
 [19] bit64_0.9-7              glue_1.3.1               sctransform_0.2.1       
 [22] processx_3.4.2           vipor_0.4.5              DOSE_3.12.0             
 [25] haven_2.2.0              tidyselect_1.0.0         usethis_1.5.1           
 [28] fitdistrplus_1.0-14      XML_3.99-0.3             zoo_1.8-7               
 [31] xtable_1.8-4             magrittr_1.5             bibtex_0.4.2.2          
 [34] Rdpack_0.11-1            cli_2.0.2                zlibbioc_1.32.0         
 [37] sn_1.5-5                 rstudioapi_0.11          miniUI_0.1.1.1          
 [40] fastmatch_1.1-0          shiny_1.4.0.1            BiocSingular_1.2.2      
 [43] pkgbuild_1.0.6           multtest_2.42.0          cluster_2.1.0           
 [46] caTools_1.18.0           tidygraph_1.1.2          ggrepel_0.8.1           
 [49] listenv_0.8.0            png_0.1-7                future_1.16.0           
 [52] withr_2.1.2              bitops_1.0-6             slam_0.1-47             
 [55] ggforce_0.3.1            cellranger_1.1.0         GSEABase_1.48.0         
 [58] sparsesvd_0.2            pillar_1.4.3             RcppParallel_4.4.4      
 [61] gplots_3.0.3             multcomp_1.4-12          fs_1.3.1                
 [64] europepmc_0.3            DelayedMatrixStats_1.8.0 vctrs_0.2.3             
 [67] ellipsis_0.3.0           generics_0.0.2           urltools_1.7.3          
 [70] devtools_2.2.2           metap_1.3                tools_3.6.3             
 [73] feather_0.3.5            beeswarm_0.2.3           munsell_0.5.0           
 [76] tweenr_1.0.1             fgsea_1.12.0             fastmap_1.0.1           
 [79] HSMMSingleCell_1.6.0     compiler_3.6.3           pkgload_1.0.2           
 [82] httpuv_1.5.2             sessioninfo_1.1.1        plotly_4.9.2            
 [85] GenomeInfoDbData_1.2.2   lattice_0.20-38          mutoss_0.1-12           
 [88] later_1.0.0              jsonlite_1.6.1           docopt_0.6.1            
 [91] graph_1.64.0             pbapply_1.4-2            lazyeval_0.2.2          
 [94] promises_1.1.0           R.utils_2.9.2            sandwich_2.5-1          
 [97] Rtsne_0.15               uwot_0.1.5               igraph_1.2.4.2          
[100] survival_3.1-8           numDeriv_2016.8-1.1      plotrix_3.7-7           
[103] htmltools_0.4.0          memoise_1.1.0            graphlayouts_0.5.0      
[106] viridisLite_0.3.0        digest_0.6.25            assertthat_0.2.1        
[109] mime_0.9                 rappdirs_0.3.1           densityClust_0.3        
[112] npsurv_0.4-0             RSQLite_2.2.0            future.apply_1.4.0      
[115] lsei_1.2-0               remotes_2.1.1            blob_1.2.1              
[118] R.oo_1.23.0              labeling_0.3             fastICA_1.2-2           
[121] RCurl_1.98-1.1           broom_0.5.5              hms_0.5.3               
[124] modelr_0.1.6             colorspace_1.4-1         BiocManager_1.30.10     
[127] mnormt_1.5-6             ggbeeswarm_0.6.0         SDMTools_1.1-221.2      
[130] Rcpp_1.0.3               RANN_2.6.1               mvtnorm_1.0-12          
[133] fansi_0.4.1              grid_3.6.3               ggridges_0.5.2          
[136] lifecycle_0.2.0          formatR_1.7              zip_2.0.4               
[139] TFisher_0.2.0            gdata_2.18.0             testthat_2.3.1          
[142] DO.db_2.9                qvalue_2.18.0            desc_1.2.0              
[145] RcppAnnoy_0.0.14         TH.data_1.0-10           htmlwidgets_1.5.1       
[148] polyclip_1.10-0          triebeard_0.3.0          gridGraphics_0.5-0      
[151] rvest_0.3.5              globals_0.12.5           codetools_0.2-16        
[154] lubridate_1.7.4          GO.db_3.10.0             FNN_1.1.3               
[157] prettyunits_1.1.1        dbplyr_1.4.2             R.methodsS3_1.8.0       
[160] gtable_0.3.0             tsne_0.1-3               DBI_1.1.0               
[163] httr_1.4.1               KernSmooth_2.23-16       stringi_1.4.6           
[166] progress_1.2.2           reshape2_1.4.3           farver_2.0.3            
[169] annotate_1.64.0          viridis_0.5.1            xml2_1.2.2              
[172] combinat_0.0-8           rvcheck_0.1.8            BiocNeighbors_1.4.2     
[175] ggplotify_0.0.5          bit_1.1-15.2             ggraph_2.0.1            
[178] pkgconfig_2.0.3          gbRd_0.4-11    `
```

**Comment by chlee-tabin** [comment]
Hi, this is a known bug. https://github.com/satijalab/seurat/issues/2438

It should be resolved by upgrading to 3.1.4.

**Comment by chlee-tabin** [comment]
JFYI, also check out https://github.com/satijalab/seurat/pull/2440#issuecomment-599565849 

`method = "matrix"`, the default, should work for reasonably large datasets (and fast), for really large datasets where casting becomes really difficult, `method="igraph" weights=T`, although NOT the intended usage, produces intended results.
